lib/src/client.dart

```diff
@@ -955,13 +957,63 @@ class Client extends MatrixApi {
   final Map<String, ProfileInformation> _profileRoomsCache = {};
   final Map<String, ProfileInformation> _profileServerCache = {};
 
+  /// Get the combined profile information for this user. First checks for a
+  /// non outdated cached profile before requesting from the server. Cached
+  /// profiles are outdated if they have been cached in a time older than the
+  /// [maxCacheAge] or they have been marked as outdated by an event in the
+  /// sync loop.
+  /// In case of an
+  ///
+  /// [userId] The user whose profile information to get.
+  @override
+  Future<CachedProfileInformation> getUserProfile(
+    String userId, {
+    Duration timeout = const Duration(seconds: 30),
+    Duration maxCacheAge = const Duration(days: 1),
+  }) async {
+    final cachedProfile = await database?.getUserProfile(userId);
+    if (cachedProfile != null &&
+        (cachedProfile.outdated != true ||
+            DateTime.now().difference(cachedProfile.updated) >= maxCacheAge)) {
+      return cachedProfile;
+    }
+
+    late final ProfileInformation profile;
```

Comment:
does this need to be late?

Reply:
Actually no. I removed the `late` :D thanks